vllm/model_executor/guided_decoding/outlines_logits_processors.py

```diff
@@ -89,7 +90,10 @@ def __call__(self, input_ids: List[int],
         allowed_tokens = allowed_tokens.masked_select(
             allowed_tokens < scores.shape[-1])
         mask.index_fill_(0, allowed_tokens, 0)
-        scores.add_(mask)
+        if current_platform.is_hpu():
+            scores = scores.add(mask)
+        else:
+            scores.add_(mask)
```

Comment:
Could we simply use `scores = scores.add(mask)` for all cases? I am not familiar with performance differences between this and `scores.add_(mask)`